### ISSUE TITLE
Add narrative documentation for a fully working twisted.names.client script

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6969]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)| @wallrj reported|
|-|-|
|Trac ID|trac#6969|
|Type|enhancement|
|Created|2014-02-12 13:30:19Z|
|Branch|https://github.com/twisted/twisted/tree/reverse-lookup-example-6969|

The client-tour howto, doesn't explain how to deal with the records returned from IResolver.lookup methods.

 * http://twisted.readthedocs.org/en/latest/projects/names/howto/client-tour.html

Add a new section with an expanded version of the reverse-lookup example from the [ProposedWikiStart](https://twistedmatrix.com/trac/wiki/ProposedWikiStart) and add some narrative introducing reverse DNS, explaining how the example works and explaining the capabilities and possible improvements to the example. eg

 * Fast, because all queries sent in parallel, but
 * Could use coiterate to limit the parallelism

<details><summary>Searchable metadata</summary>

```
trac-id__6969 6969
type__enhancement enhancement
reporter__rwall rwall
priority__normal normal
milestone__ 
branch__branches_reverse_lookup_example_6969 branches/reverse-lookup-example-6969
branch_author__rwall rwall
status__closed closed
resolution__fixed fixed
component__names names
keywords__documentation documentation
time__1392211819000000 1392211819000000
changetime__1393239616000000 1393239616000000
version__None None
owner__rwall rwall

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_1"></a>|@wallrj commented|
|-|-|

(In [41663]) Branching to 'reverse-lookup-example-6969'

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_2"></a>|@wallrj removed owner|
|-|-|

Ready for review in log:branches/reverse-lookup-example-6969:
 * Added a new "simple example" section to the client-tour howto
 * With narrative to the simple reverse DNS lookup example which is proposed for the wiki frontpage ([ProposedWikiStart](https://twistedmatrix.com/trac/wiki/ProposedWikiStart))
 * Also added a more complete reverse DNS lookup example which performs both IPv4 and IPv6 reverse lookups in parallel

Build Results:
 * https://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/reverse-lookup-example-6969

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set owner to @wallrj|
|-|-|

Wow.  Tests for the doc examples and everything!  This looks great.

Please address the one twistedchecker error and land.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_4"></a>|@wallrj set status to `closed`|
|-|-|

(In [41744]) Merge reverse-lookup-example-6969

Author: rwall
Reviewer: glyph
Fixes: [#6969](https://github.com/twisted/twisted/issues/6969)

twisted.names now has narrative documentation and examples showing how to
perform reverse DNS lookups.
